### ISSUE TITLE
bug: Live Reloading doesn't work at 5.8.0 but seems 5.7.0 is okay.

### ISSUE THREAD

**Issue by sunyuzu** [ionic-team/ionic-framework#23963]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

When I install a new ionic framework project(v5.8.0), `ionic serve` live reload is not working. So I've tried to find the problem on my vue source codes, it didn't work. However my old ionic project has been okay(v5.7.0). I've checked all files one by one then I've found it is okay when I copied `package-lock.json` and `package.json` from v5.7.0 to v5.8.0 then npm install again. There are many changes regarding Vue packages from v3.2.8 to v3.2.12. I have doubts with this.
I hope to fix this bug for all Vue users.

You can find all behavior with git animation [here](https://d.pr/i/aaknGV).

*Browser Errors*:
### in Firefox
![](https://i.esdrop.com/d/richard_salt/iovi21yCk4.png)
### in Chrome
![](https://i.esdrop.com/d/richard_salt/xNERpn2UYP.png)
### in Safari
![](https://i.esdrop.com/d/richard_salt/Dmg5crrsQf.png)

### Expected Behavior

Expect no errors...

### Steps to Reproduce

Just as the quick installation guide.
```
$ npm install -g @ionic/cli@latest
$ ionic start myApp blank --type vue
$ cd myApp
$ ionic serve
```
Then modify some codes and check the browser live reloading.

### @ionic/vue 5.8.0 (not work)
```
Ionic:

   Ionic CLI       : 6.17.1 (/Users/richard/.nvm/versions/node/v16.5.0/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 5.8.0

Capacitor:

   Capacitor CLI      : 3.2.3
   @capacitor/android : 3.2.3
   @capacitor/core    : 3.2.3
   @capacitor/ios     : 3.2.3

Utility:

   cordova-res : 0.15.3
   native-run  : 1.4.1

System:

   NodeJS : v16.5.0 (/Users/richard/.nvm/versions/node/v16.5.0/bin/node)
   npm    : 7.21.1
   OS     : macOS Big Sur
```

### Additional Information

When I've copied `package-lock.json` and `package.json` from v5.7.0 to v5.8.0, it worked.

### @ionic/vue 5.7.0 (worked)
```
Ionic:

   Ionic CLI       : 6.17.1 (/Users/richard/.nvm/versions/node/v16.5.0/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 5.7.0

Capacitor:

   Capacitor CLI      : 3.2.2
   @capacitor/android : 3.2.2
   @capacitor/core    : 3.2.2
   @capacitor/ios     : 3.2.2

Utility:

   cordova-res : 0.15.3
   native-run  : 1.4.1

System:

   NodeJS : v16.5.0 (/Users/richard/.nvm/versions/node/v16.5.0/bin/node)
   npm    : 7.21.1
   OS     : macOS Big Sur
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. I can reproduce this even if I remove Ionic. Let me dig into this to see what is going on.

**Comment by liamdebeasi** [comment]
Looks like other Vue 3 devs have been running into similar issues: https://forum.vuejs.org/t/vue-3-fresh-project-hot-reload-resulting-in-component-is-missing-template-or-render-function/121884/3

**Comment by liamdebeasi** [comment]
Looks to be a Vue bug. I opened an issue here: https://github.com/vuejs/vue-next/issues/4654

**Comment by sunyuzu** [comment]
Evan You mentioned [vuejs/vue-next#4654](https://github.com/vuejs/vue-next/issues/4654),

> Should be fixed in vue-loader@16.8.1.

So I've tried fresh install for ionic, it's back! And I've found `npm:vue-loader@16.8.1` in `package-lock.json`,
```
...
    "vue-loader-v16": {
      "version": "npm:vue-loader@16.8.1", 👈👈👈
      "resolved": "https://registry.npmjs.org/vue-loader/-/vue-loader-16.8.1.tgz",
      "integrity": "sha512-V53TJbHmzjBhCG5OYI2JWy/aYDspz4oVHKxS43Iy212GjGIG1T3EsB3+GWXFm/1z5VwjdjLmdZUFYM70y77vtQ==",
      "dev": true,
      "optional": true,
...
```
I feel little bit more blinking than before when reloading. It's more like re-loading but HMR. but it seems from new Vue-loader, hope will update. Now I can enjoy coding again, thank you.

**Comment by liamdebeasi** [comment]
Glad the issue is resolved! I am going to close this issue out. Thanks for reporting this!
